modules/control/controller/mpc_controller.cc

```diff
@@ -325,34 +328,66 @@ Status MPCController::ComputeControlCommand(
   debug->add_matrix_r_updated(matrix_r_updated_(0, 0));
   debug->add_matrix_r_updated(matrix_r_updated_(1, 1));
 
-  Eigen::MatrixXd control_matrix(controls_, 1);
-  control_matrix << 0, 0;
+  Matrix control_matrix = Matrix::Zero(controls_, 1);
+  std::vector<Matrix> control(horizon_, control_matrix);
+
+  Matrix control_gain_matrix = Matrix::Zero(controls_, basic_state_size_);
+  std::vector<Matrix> control_gain(horizon_, control_gain_matrix);
 
-  Eigen::MatrixXd reference_state(basic_state_size_, 1);
-  reference_state << 0, 0, 0, 0, 0, 0;
+  Matrix addition_gain_matrix = Matrix::Zero(controls_, 1);
+  std::vector<Matrix> addition_gain(horizon_, addition_gain_matrix);
 
-  std::vector<Eigen::MatrixXd> reference(horizon_, reference_state);
+  Matrix reference_state = Matrix::Zero(basic_state_size_, 1);
+  std::vector<Matrix> reference(horizon_, reference_state);
 
-  Eigen::MatrixXd lower_bound(controls_, 1);
+  Matrix lower_bound(controls_, 1);
   lower_bound << -wheel_single_direction_max_degree_, max_deceleration_;
 
-  Eigen::MatrixXd upper_bound(controls_, 1);
+  Matrix upper_bound(controls_, 1);
   upper_bound << wheel_single_direction_max_degree_, max_acceleration_;
 
-  std::vector<Eigen::MatrixXd> control(horizon_, control_matrix);
-
   double mpc_start_timestamp = Clock::NowInSeconds();
   double steer_angle_feedback = 0.0;
   double acc_feedback = 0.0;
+  double steer_angle_ff_compensation = 0.0;
+  double unconstraint_control_diff = 0.0;
+  double unconstraint_control_diff_limit = 5.0;
```

Comment:
make it a control conf in the controller configuration.  Append it here https://github.com/ApolloAuto/apollo/blob/master/modules/control/proto/mpc_controller_conf.proto